### PR TITLE
REL-899797 update readme

### PR DESCRIPTION
**Pull request by KlaudiuszBryjaRelativity**

### CODE REVIEW

**Review by dj-wyder — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -15,68 +18,81 @@ This project showcases a straightforward integration scenario with `TransferSDK`
...
+- `Relativity.Transfer.SDK.Samples.Core` - contains all the interfaces, models, UI, and helpers used by the `Relativity.Transfer.SDK.Samples.Repository` project.
+- `Relativity.Transfer.SDK.Samples.Repository` - contains all the samples.
+- `Relativity.Transfer.SDK.Samples.CLI` - contains a command-line interface for the `Relativity.Transfer.SDK.Samples.Repository` project. It uses IoC container to create instances of required classes.
+
 #### Examples structure:
-- Each sample is numbered. The number of a sample is included in the file and the class name.
-- All code presenting a particular example is contained only in a single file (if additional classes are needed for the example they are defined as private within the example).
-- Usually, all necessary inputs are taken at the beginning of the example using `ConsoleHelper` class.
-- `ConsoleHelper` prompts a user for input only when the input was not set up before. That means after initial setup some samples can work without prompting the user at all.
+- All code presenting a particular example is contained only in a single file (if additional dependencies are required they are injected by IoC).
```
Comment (KamilKalitaRel) [inline README.md:28]: Using of IoC contradicts the statement that "everything is one file". Maybe:
```suggestion
- Code presenting a particular example is contained in a single file (see the table below to match a sample with a code file). If additional dependencies are required they are injected by IoC [here the rough location, when this IoC is defined].
```

File: README.md [hunk 2]
```diff
@@ -15,68 +18,81 @@ This project showcases a straightforward integration scenario with `TransferSDK`
...
-        - The password is not stored in the `App.config` file by default
-        - To store the password the `App.config` must be directly modified, under the `ClientPassword` section.
-- There are 3 options to edit the above settings:
-    - during application runtime via the provided console interface
-    - by modifying the application *.config file in the bin folder.
-    - by modifying `App.config` file and rebuilding the application (and cleaning the `obj` folder)
-- To be prompted for the settings every time the operation is made, the required settings in the config should be empty.
+        - **Note**: It is crucial to remember that backslash `\` characters should be escaped using a double backslash `\\` (this is required by json file to properly deserialize a path).
+    - `Common.OAuthCredentials` - The OAuth2 secret ID and secret used to authenticate in RelativityOne (Find the `OAuth2 Client` tab, then find the user and copy `ClientId` and `ClientSecret` fileds).
+- The sttings specified for a sample are commented in `appsettings.json` file. Which section is responsible for which sample is described in the table above.
```
Comment (KamilKalitaRel) [inline README.md:73]:
```suggestion
- The settings specified for a sample are commented in `appsettings.json` file. Which section is responsible for which sample is described in the table above.
```

File: README.md [hunk 3]
```diff
@@ -15,68 +18,81 @@ This project showcases a straightforward integration scenario with `TransferSDK`
...
+    .Build();
+
+// or JobBasedWorkflow
+var transferClient = TransferClientBuilder.FullPathWorkflow
+    .WithAuthentication(authenticationProvider)
+    .WithClientName(clientName)
+    .Build();
+```
+
+- `TransferSKD` uses a bearer token in order to authenticate the transfer.
```
Comment (KamilKalitaRel) [inline README.md:91]:
```suggestion
- `TransferSDK` uses a bearer token in order to authenticate the transfer.
```

File: README.md [hunk 4]
```diff
@@ -15,68 +18,81 @@ This project showcases a straightforward integration scenario with `TransferSDK`
...
+- To pass the token, the `Relativity.Transfer.SDK.Interfaces.Authentication.IRelativityAuthenticationProvider` must be implemented and an instance should be passed to the clint. Sample implementation of authentication provider is [here](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/RelativityAuthenticationProvider.cs).
```
Comment (KamilKalitaRel) [inline README.md:92]:
```suggestion
- To pass the token, the `Relativity.Transfer.SDK.Interfaces.Authentication.IRelativityAuthenticationProvider` must be implemented and an instance should be passed to the client. Sample implementation of authentication provider is [here](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/RelativityAuthenticationProvider.cs). 
```

File: README.md [hunk 5]
```diff
@@ -15,68 +18,81 @@ This project showcases a straightforward integration scenario with `TransferSDK`
...
+// or JobBasedWorkflow
+var transferClient = TransferClientBuilder.FullPathWorkflow
+    .WithAuthentication(authenticationProvider)
+    .WithClientName(clientName)
+    .Build();
+```
+
+- `TransferSKD` uses a bearer token in order to authenticate the transfer.
+- To pass the token, the `Relativity.Transfer.SDK.Interfaces.Authentication.IRelativityAuthenticationProvider` must be implemented and an instance should be passed to the clint. Sample implementation of authentication provider is [here](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/RelativityAuthenticationProvider.cs). 
+- In order to get the token, [Bearer token authentication](https://platform.relativity.com/RelativityOne/Content/REST_API/REST_API_authentication.htm#_Bearer_token_authentication) is used, which requires user OAuth2 client id and client secret. (see [SampleBearerTokenRetriver](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/BearerTokenRetriever.cs))
```
Comment (KamilKalitaRel) [inline README.md:93]:
```suggestion
- In order to get the token, [Bearer token authentication](https://platform.relativity.com/RelativityOne/Content/REST_API/REST_API_authentication.htm#_Bearer_token_authentication) is used, which requires user's OAuth2 client id and client's secret. (see [SampleBearerTokenRetriver](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/BearerTokenRetriever.cs))
```

**Review by KamilKalitaRel — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -15,74 +18,96 @@ This project showcases a straightforward integration scenario with `TransferSDK`
...
+- First, the transfer client object has to be created, which is used to manage transfers:
+```cs
+// FullPathWorkflow
+var transferClient = TransferClientBuilder.FullPathWorkflow
+    .WithAuthentication(authenticationProvider)
+    .WithClientName(clientName)
+    .Build();
+
+// or JobBasedWorkflow
+var transferClient = TransferClientBuilder.FullPathWorkflow
```
Comment (avichay-kardash) [inline README.md]: Change to JobBasedWorkflow

**Review by avichay-kardash — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (KlaudiuszBryjaRelativity) [reply README.md]: done

**Pull request merged**